### DISCUSSION TITLE
Is there a list of used ports?

### DISCUSSION THREAD

**Discussion by goloroden** [community/community#25550] [Actions]
<p>Hey there :wave:</p>
<p>I am currently experimenting with GitHub Actions, and although my first impression is pretty good, a question came up: In my tests I am building a number of Docker containers (this is actually part of the test), and after that I start them using <code>docker run</code>.</p>
<p>Basically, this works (at least on Ubuntu), but some of the ports I would like to use are already being used.</p>
<p>So, my question is: Is there a list of already used ports available somewhere?</p>
<p>Have a nice day,</p>
<p>Golo</p>

**Comment by Yanjingzhu** [comment]
<p>You can add one step to run <strong>docker ps</strong>  <strong>–a</strong> command to list all containers and in your workflow. And then run <strong>docker port CONTAINER</strong> to show all mapped ports.</p>
<p>Please refer to this docker port document: <a href="https://docs.docker.com/engine/reference/commandline/port/" rel="nofollow noopener">https://docs.docker.com/engine/reference/commandline/port/</a></p>
<p>What’s the exact error message you got?  </p>
<p>For us to understand your scenario more clearly, could you please share your workflow yml content and workflow run logs here? If you use a public repo, could you provide your workflow link? </p>

**Comment by goloroden** [comment]
<p>Hi Yanjingzhu!</p>
<p>Thanks for your answer!</p>
<p>The repository is actually public, and you can find it (including the GitHub actions branch) here: <a href="https://github.com/thenativeweb/wolkenkit/tree/introduce-github-actions" rel="nofollow noopener">https://github.com/thenativeweb/wolkenkit/tree/introduce-github-actions</a></p>
<p>The concrete error message is:</p>
<p><strong>Error starting userland proxy: listen tcp 0.0.0.0:3306: bind: address already in use.</strong></p>
<p>The appropriate build is this one: <a href="https://github.com/thenativeweb/wolkenkit/commit/96c715cba2657cdb4f71bc16fc6b0ffe2db28685/checks?check_suite_id=324771001" rel="nofollow noopener">https://github.com/thenativeweb/wolkenkit/commit/96c715cba2657cdb4f71bc16fc6b0ffe2db28685/checks?check_suite_id=324771001</a></p>
<p>Thanks for the hint with docker ps, but I was hoping for an official list (maybe not all currently free ports are available in the long run), but it’s a way to go…</p>
<p>Looking forward to your feedback and ideas :slight_smile:</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Thank you for your workfkow yml and logs. 3306 port is the default port which used by MySQL. MySQL is installed on Github actions Ubuntu runner.</p>
<p><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#ubuntu-1804-lts" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#ubuntu-1804-lts</a>  </p>
<p>So before set-up a MySQL which host port is 3306 in Docker, please make sure the Default MySQL has been shutted-down.</p>
<p>Please add a step before Run roboter to shutdown the Default MySQL:</p>
<pre><code>    - name: Shutdown Ubuntu MySQL (SUDO)
      run: sudo service mysql stop # Shutdown the Default MySQL, "sudo" is necessary, please not remove it
</code></pre>
<p>For more information, please refer to <a href="https://github.com/mirromutth/mysql-action#shutdown-the-default-mysql" rel="nofollow noopener">https://github.com/mirromutth/mysql-action#shutdown-the-default-mysql</a></p>

**Comment by goloroden** [comment]
<p>Thanks a lot :slight_smile:</p>
